eclass/java-utils-2.eclass

```diff
@@ -2945,7 +2943,14 @@ is-java-strict() {
 java-pkg_clean() {
 	NO_DELETE=()
 	for keep in ${JAVA_PKG_NO_CLEAN[@]}; do
-		NO_DELETE+=( '!' '-wholename' ${keep} )
+		if [[ "${keep%.jar}" != ${keep} ]]; then
+			NO_DELETE+=( '-type f' '!' '-name' ${keep} )
```

Comment:
i didn't test but i think you cannot specify `-name ./libs/standard.jar` as it contains also a path, not just the file name

Reply:
You are right, that doesn't work. The focus of the proposed change is allowing file names. These cases are identified with parameter expansion `${keep%.jar}`.  For `paths` I haven't yet found a solution.  

Expressions would still be possible and specifying pure file names would be simplified. 

Specifying a file in a certain directory ( a path ending on `.class` or `.jar` ) would still be tricky.  It would work with a trailing asterisk and in quotes:
```
JAVA_PKG_NO_CLEAN=(
	"*/all/wrapper.jar*"
)
```
Following output is from upcoming `net-vpn/i2p-2.1.0` which has 3 `*wrapper.jar` files. Please regard this only as an example for demonstration - I don't want to keep those files! 
```
Search done.
 * keeping standard.jar
 * keeping */all/wrapper.jar*
./apps/susidns/src/lib/jstl.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/start.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-servlet-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-util-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-xml-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-rewrite-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-security-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/apache-jsp/org.eclipse.jetty.apache-jsp-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-continuation-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-webapp-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-io-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-jmx-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-server-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-servlets-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-deploy-9.3.30.v20211001.jar
./apps/jetty/jetty-distribution-9.3.30.v20211001/lib/jetty-http-9.3.30.v20211001.jar
./apps/jetty/apache-tomcat-9.0.62/bin/tomcat-juli.jar
./apps/jetty/apache-tomcat-9.0.62/lib/servlet-api.jar
./apps/jetty/apache-tomcat-9.0.62/lib/jasper.jar
./apps/jetty/apache-tomcat-9.0.62/lib/jsp-api.jar
./apps/jetty/apache-tomcat-9.0.62/lib/tomcat-util.jar
./apps/jetty/apache-tomcat-9.0.62/lib/tomcat-api.jar
./apps/jetty/apache-tomcat-9.0.62/lib/tomcat-util-scan.jar
./apps/jetty/apache-tomcat-9.0.62/lib/el-api.jar
./apps/jetty/apache-tomcat-9.0.62/lib/jasper-el.jar
./installer/lib/launch4j/launch4j.jar
./installer/lib/launch4j/lib/xstream.jar
./installer/lib/launch4j/lib/looks.jar
./installer/lib/launch4j/lib/forms.jar
./installer/lib/launch4j/lib/foxtrot.jar
./installer/lib/launch4j/lib/commons-beanutils.jar
./installer/lib/launch4j/lib/commons-logging.jar
./installer/lib/launch4j/lib/formsrt.jar
./installer/lib/izpack/standalone-compiler.jar
./installer/lib/wrapper/win-all/wrapper.jar
./gradle/wrapper/gradle-wrapper.jar
>>> Source prepared.
```
I'm still trying to find a way to improve handling of `paths` but I think the proposed simplification for file names is a step in the right direction.